packages/twenty-server/src/engine/seeder/seeder.service.ts

```diff
@@ -0,0 +1,161 @@
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  public async seedCustomObjects(
+    dataSourceId: string,
+    workspaceId: string,
+    metadataSeeds: ObjectMetadataSeed,
+    dataSeeds: Record<string, any>[],
+  ): Promise<void> {
+    const createdObjectMetadata = await this.objectMetadataService.createOne({
+      ...metadataSeeds,
+      dataSourceId,
+      workspaceId,
+    });
+
+    if (!createdObjectMetadata) {
+      throw new Error("Object metadata couldn't be created");
+    }
+
+    for (const customField of metadataSeeds.fields) {
+      await this.fieldMetadataService.createOne({
+        ...customField,
+        objectMetadataId: createdObjectMetadata.id,
+        workspaceId,
+      });
+    }
+
+    const objectMetadataAfterFieldCreation =
+      await this.objectMetadataService.findOneWithinWorkspace(workspaceId, {
+        where: { nameSingular: metadataSeeds.nameSingular },
+      });
+
+    if (!objectMetadataAfterFieldCreation) {
+      throw new Error(
+        "Object metadata couldn't be found after field creation.",
+      );
+    }
+
+    const schemaName =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const workspaceDataSource =
+      await this.workspaceDataSourceService.connectToWorkspaceDataSource(
+        workspaceId,
+      );
+
+    const entityManager = workspaceDataSource.createEntityManager();
+
+    const filteredFields = metadataSeeds.fields.filter((field) =>
+      objectMetadataAfterFieldCreation.fields.some(
+        (f) => f.name === field.name || f.name === `name`,
+      ),
+    );
+
+    if (filteredFields.length === 0) {
+      throw new Error('No fields found for seeding, check metadata');
+    }
+
+    filteredFields.unshift({
+      name: 'name',
+      type: FieldMetadataType.TEXT,
+      label: 'Name',
+    });
+
+    const fieldMetadataMap = filteredFields
+      .map((field) => {
+        if (isCompositeFieldMetadataType(field.type)) {
+          const compositeFieldTypeDefinition = compositeTypeDefinitions.get(
+            field.type,
+          );
+
+          if (!isDefined(compositeFieldTypeDefinition)) {
+            throw new Error(
+              `Composite field type definition not found for ${field.type}`,
+            );
+          }
+
+          const fieldNames = compositeFieldTypeDefinition.properties?.map(
+            (property) => property.name,
+          );
+
+          return (
+            fieldNames?.map(
+              (subFieldName: string) =>
+                `${field.name}${capitalize(subFieldName)}`,
+            ) ?? []
+          );
+        } else {
+          return field.name;
+        }
+      })
+      .flat()
+      .filter(isDefined);
+
+    const flattenedSeeds = dataSeeds.map((seed) => {
+      const flattenedSeed = {};
+
+      for (const field of filteredFields) {
+        if (isCompositeFieldMetadataType(field.type)) {
```

Comment:
nit: Could be refactored, looks like we are doing the same checks here and above